tests/Integration/Integrations/IntegrationsTest.php

```diff
@@ -20,6 +21,15 @@
  * @todo: Instantiate and then try to register something that doesn't implement the Integration interface.
  */
 final class IntegrationsTest extends TestCase {
+	/**
+	 * The setUpBeforeClass run before all tests
+	 */
+	public static function setUpBeforeClass(): void {
+		parent::setUpBeforeClass();
+
+		$GLOBALS['parsely'] = new Parsely();
+	}
```

Comment:
Why is this change needed?

Reply:
As per the changes in `wp-parsely.php`, the Web Stories class needs a Parse.ly object to be initialized. We take that object from `$GLOBALS` as we do with all the other classes we init there.

Reply:
The issue is in line 52 of `IntegrationsTest`. That calls a function on `wp-parsely.php`, `parsely_integrations` which instantiates GWS class on line 137. That constructor is called with the global variable. That global variable is always present on normal execution because it's created in line 60 of the same file, but that hook doesn't get executed on tests.

If I remove the global declaration on the test, tests fail. I agree though that using globals on tests is not the nicest.

Reply:
I understand now - thanks for the explanation.

Can you please add a comment to this line 29 then, to explain the reasoning? Something like:

```php
// Add a global instance, so that calls to `parsely_integrations()` in tests can correctly instantiate and register Google_Web_Stories as an integration.
```